sdk/identity/azure-identity/test/ut/managed_identity_credential_test.cpp

```diff
@@ -785,6 +801,50 @@ TEST(ManagedIdentityCredential, CloudShellInvalidUrl)
       {"{\"expires_in\":3600, \"access_token\":\"ACCESSTOKEN1\"}"}));
 }
 
+// This helper creates the necessary directories required for the Azure Arc tests, and returns where
+// we expect the valid arc key to exist.
+std::string CreateDirectoryAndGetValidKeyPath()
+{
+  std::string validKeyPath;
+#if defined(AZ_PLATFORM_LINUX)
+  validKeyPath = "/var/opt/azcmagent/tokens";
+  int result = system(std::string("sudo mkdir -p ").append(validKeyPath).c_str());
```

Comment:
Will this work when run interactively on Linux? 

Under both ctest and from gtest?

Reply:
I am checking that now, at least with ctest. Looks like there are some permissions limitations on the CI machines.

Not sure how to run that with gtest, explicitly.

Reply:
gtest means running the test explicitly from the command line. ctest means running it from ctest in the build directory.

Each of the test mechanisms behaves differently w.r.t. stdin and stdout (ctest redirects stdin and stdout to a file, gtest just leaves things alone - that means that I suspect that if sudo is going to prompt, it will work from gtest and fail from ctest). 

IMHO tests should always work when run from a developer machine.

Reply:
It does run locally, both with ctest and gtest.

This prompts for the password, in interactive Linux session
> test -R azure-identity.ManagedIdentityCredential.AzureArc

This runs with the typical gtest flow.
>  ./sdk/identity/azure-identity/test/ut/azure-identity-test